scipy/sparse/sputils.py

```diff
@@ -251,6 +251,88 @@ def validateaxis(axis):
             raise ValueError("axis out of range")
 
 
+def check_shape(args, current_shape=None):
+    """Imitate numpy.matrix handling of shape arguments"""
+    def check_int(arg):
```

Comment:
I think you can replace this function with `operator.index(arg)`. (See `sputils.isintlike`.)

Reply:
Yep, `operator.index` what I was looking for. It seems to be completely consistent with how `numpy.matrix` handles them, like I intended. I see that you recently updated `sputils.isinitlike` to use it.